modules/apps/redirect/redirect-test/src/testIntegration/java/com/liferay/redirect/web/internal/instance/lifecycle/test/RedirectEntryPermissionPortalInstanceLifecycleListenerTest.java

```diff
@@ -0,0 +1,146 @@
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.redirect.web.internal.instance.lifecycle.test;
+
+import com.liferay.arquillian.extension.junit.bridge.junit.Arquillian;
+import com.liferay.portal.instance.lifecycle.PortalInstanceLifecycleListener;
+import com.liferay.portal.kernel.model.Company;
+import com.liferay.portal.kernel.model.Group;
+import com.liferay.portal.kernel.model.ResourceConstants;
+import com.liferay.portal.kernel.model.Role;
+import com.liferay.portal.kernel.model.role.RoleConstants;
+import com.liferay.portal.kernel.security.permission.ActionKeys;
+import com.liferay.portal.kernel.service.GroupLocalService;
+import com.liferay.portal.kernel.service.ResourcePermissionLocalService;
+import com.liferay.portal.kernel.service.RoleLocalService;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.rule.DeleteAfterTestRun;
+import com.liferay.portal.kernel.test.util.CompanyTestUtil;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.redirect.model.RedirectEntry;
+
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.service.component.runtime.ServiceComponentRuntime;
+import org.osgi.service.component.runtime.dto.ComponentDescriptionDTO;
+import org.osgi.util.promise.Promise;
+
+/**
+ * @author tototrinh
+ */
+@RunWith(Arquillian.class)
+public class RedirectEntryPermissionPortalInstanceLifecycleListenerTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Test
+	public void testViewRedirectEntryPermissionsAddedWithExistingCompany()
+		throws Exception {
+
+		Class<?> clazz = _portalInstanceLifecycleListener.getClass();
+
+		ComponentDescriptionDTO componentDescriptionDTO =
+			_serviceComponentRuntime.getComponentDescriptionDTO(
+				FrameworkUtil.getBundle(clazz), clazz.getName());
+
+		_disablePortalInstanceLifecycleListener(componentDescriptionDTO);
+
+		_testCompany = CompanyTestUtil.addCompany();
+
+		Assert.assertFalse(
+			_hasViewRedirectEntryPermission(_testCompany.getCompanyId()));
+
+		_enablePortalInstanceLifecycleListener(componentDescriptionDTO);
+
+		Assert.assertTrue(
+			_hasViewRedirectEntryPermission(_testCompany.getCompanyId()));
```

Comment:
When changing the state of the server in integration tests, you need to make sure you bring it back to its original state even in the case of failure. In this case, if an assertion fails, or some other error occurs before the _enable* method is executed, you'll leave the component disabled.
In this case, the fix is to wrap the tests in a try/finally block:
```
try {
    ...
    _disablePortalInstanceLifecycleListener(...);
    ...
}
finally {
    _enablePortalInstanceLifecycleListener(...);
}
```
Can you apply this change?

Reply:
My bad, I will update the code. Thanks.